AEPNotificationContent/Sources/AEPNotificationViewController.swift

```diff
@@ -14,48 +14,93 @@ import UIKit
-            if let payload = BasicPayload(from: notification.request.content) {
-                let basicTemplateController = BasicTemplateController(payload)
-                view.addSubview(basicTemplateController.view)
-                preferredContentSize.height = basicTemplateController.preferredContentSize.height
-            } else {
-                displayFallbackTemplate(notification)
-            }
-        case AEPNotificationContentConstants.PayloadKey.TemplateType.CAROUSEL:
-            print("//TODO: Create and display Carousel viewController")
-        case AEPNotificationContentConstants.PayloadKey.TemplateType.TIMER:
-            print("//TODO: Create and display Timer viewController")
-        default:
-            displayFallbackTemplate(notification)
+        // Create a controller based on the template type
+        let controller: UIViewController?
+        if let basicPayload = payload as? BasicPayload {
+            controller = BasicTemplateController(withPayload: basicPayload, delegate: self)
+        } else { controller = nil }
+
+        // If a controller was created, add it as a child and its view to the hierarchy
+        if let controller = controller {
+            addChild(controller)
+            view.addSubview(controller.view)
+        } else {
+            displayFallbackTemplate(notificationContent)
         }
     }
 
     public func didReceive(_: UNNotificationResponse, completionHandler _: @escaping (UNNotificationContentExtensionResponseOption) -> Void) {}
 
+    // MARK: - UIViewController delegate methods
+
+    /// Called after the controller's view is loaded into memory
     override public func viewDidLoad() {
         super.viewDidLoad()
         view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
     }
 
+    /// Called when the preferred content size of one of the view controller's child content containers changes
     override open func preferredContentSizeDidChange(forChildContentContainer container: UIContentContainer) {
         preferredContentSize.height = container.preferredContentSize.height
     }
 
-    private func displayFallbackTemplate(_ notification: UNNotification) {
-        let fallback = FallbackTemplateController(notificationContent: notification.request.content)
+    // MARK: - Private methods
+
+    /// Create a payload object based on the template type
+    /// Returns nil if the template type is not supported or if the payload object could not be created
+    /// - Parameters:
+    ///   - templateType: String representing the template type
+    ///   - content: UNNotificationContent object from received notification
+    /// - Returns: the appropriate Payload object
+    private func createPayload(for templateType: String, from content: UNNotificationContent) -> Payload? {
+        switch templateType {
+        case AEPNotificationContentConstants.PayloadKey.TemplateType.BASIC:
+            return BasicPayload(from: content)
+        default:
```

Comment:
let's make this an `@unknown default`

Reply:
Reading about @unknown. Looks like they only work on enum that are nonfrozen
We are using string value in this switch and I cannot use @unknown 

And changing string to enum looks like an extra step at this point. I will consider converting templateType as an enum in the future, If I need to use them in multiple places.

Reply:
yep, i'm an idiot :D